subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5718,43 +5719,34 @@ int hci_iso_handle(struct net_buf *buf, struct net_buf **evt)
 		}
 
 		struct ll_conn_iso_group *cig = cis->group;
+		uint8_t event_offset;
 
 		hdr = &(cis->hdr);
 
-		/* Set target event as the current event. This might cause some
-		 * misalignment between SDU interval and ISO interval in the
-		 * case of a burst from the application or late release. However
-		 * according to the specifications:
-		 * BT Core V5.3 : Vol 6 Low Energy Controller : Part B LL Spec:
-		 * 4.5.13.3 Connected Isochronous Data:
-		 * This burst is associated with the corresponding CIS event but
-		 * the payloads may be transmitted in later events as well.
-		 * If flush timeout is greater than one, use the current event,
-		 * otherwise postpone to the next.
+		/* We must ensure sufficient time for ISO-AL to fragment SDU and
+		 * deliver PDUs to the TX queue. By checking ull_ref_get, we
+		 * know if we are within the subevents of an ISO event. If so,
+		 * we can assume that we have enough time to deliver in the next
+		 * ISO event. If we're not active within the ISO event, we don't
+		 * know if there is enough time to deliver in the next event,
+		 * and for safety we set the target to current event + 2.
```

Comment:
Is it a timing issue? 

i.e. we know we can complete it in the timespan on an event, but since we don't know how close we are to the end of the current event, we set the end of the next event as the deadline?

Reply:
Yes. If we target current event +1 we might be too close to the start of the event and the fragmented PDUs might not be ready in time for transmission.

Reply:
OK, thanks, now I understand it :)